src/index.ts

```diff
@@ -68,7 +70,13 @@ export const release = async (
   return result;
 };
 
-release()
+const configFile = processInputConfigFile();
```

Comment:
I can't comment on the relevant line, but in this file, on line 61, we are defining plugin specific properties at the root level of the semantic-release configuration. If a plugin were to be added by a user that would make use of one of this properties and the added configuration omits defining it, the root level value will be used. 

I think that this would provide an unexpected result.

We should explicitly pass these plugin configuration options to the `generatePlugins` function: `preset`, `releaseRules`, `writerOpts`, `parserOpts`.

Reply:
This is what `semantic-release` does itself, that's why I added the link to the docs saying it behaves the same as writing an rc file for `semantic-release`. Wdyt?

Reply:
I think we are talking about two different things. Our semantic-release configuration includes plugin specific properties that are only picked up if the plugins are included in the configuration. My suggestion was to move those plugin specific properties into the plugin declarations themselves (`generatePlugins`)and keep the root level semantic-release configuration clean so as to prevent the situation where these properties are being unknowingly picked up by a user's configuration.

Reply:
Gotcha! Fixed in b66d63dc67cded9c3c336ef1b00bc774ddc77dd6